src/main/resources/gichaAPI.yaml

```diff
@@ -0,0 +1,183 @@
+              schema:
+                $ref: '#/components/schemas/character'
+        400:
+          description: Invalid id
+    delete:
+      tags:
+        - Character
+      summary: Delete a Character based on id
+      parameters:
+        - in: path
+          name: id
+          description: Character id to delete
+          required: true
+          schema:
+            type: number
+      responses:
+        204:
+          description: Character deleted successfully
+        404:
+          description: Character not found
+        403:
+          description: Unauthorized user
+        401:
+          description: User does not have access rights
+  /characters:
+    get:
+      tags:
+        - Characters
+      summary: Get all characters from the system
+      responses:
+        200:
+          description: Success Response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/characters'
+  /characters{birthplace}:
+    get:
+      tags:
+        - Characters
+      summary: Get all characters from any location
+      parameters:
+        - in: path
+          name: birthplace
+          required: true
+          schema:
+            type: string
+            enum: ['MONDSTADT','LIYUE','INAZUMA','SUMERU','FONTAINE','NATLAN','SNEZHNAYA']
+            example: Inazuma
+      responses:
+        200:
+          description: Successfull response with param
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/characters'
+        404:
+          description: Character not found
+  /characters{element}:
+    get:
+      tags:
+        - Characters
+      summary: Get all characters with same element
+      parameters:
+        - in: path
+          name: element
+          required: true
+          schema:
+            type: string
+            enum: ['ADAPTIVE','PYRO','CRYO','ELECTRO','HYDRO','DENDRO','GEO','ANEMO']
+            example: PYRO
+      responses:
+        200:
+          description: Successfull response with param
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/characters'
+        404:
+          description: Character not found
+  /character{birthplace}:
+    get:
+      tags:
+        - Character
+      summary: Get character by location
+      parameters:
+        - in: path
+          name: birthplace
+          required: true
+          schema:
+            type: string
+            enum: ['MONDSTADT','LIYUE','INAZUMA','SUMERU','FONTAINE','NATLAN','SNEZHNAYA']
+            example: Inazuma
+      responses:
+        404:
+          description: Character not found
+        200:
+          description: Successfull response with param
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/character'
+
+components:
+  schemas:
+    character:
```

Comment:
Do schemat (tj. do datovych modelu) se dava naopak velke pismeno, aby se vizualne odlisilo, ze jde o objekt.

Reply:
ty vole, ok. Ale pak az budu delat mapper, bude schopen poznat že name (dto classa) je to samy jako Name (schema)? Predtim byly zcela jine nazvy, takže to nefungovalo bez konkretni identifikace, ale tady?

Reply:
Podle importu, mapper si poradi ..